rackcity/views/rack_views.py

```diff
@@ -85,4 +109,48 @@ def rack_delete(request):
     """
     Delete racks within specified range.
     """
-    return JsonResponse({}, status=HTTPStatus.INTERNAL_SERVER_ERROR)
+    range_serializer = RackRangeSerializer(data=request.data)
+    if not range_serializer.is_valid():
+        return JsonResponse(
+            {"failure_message": "Invalid keys or values in request."},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    racks = Rack.objects.filter(
+        rack_num__range=range_serializer.get_number_range(),  # inclusive range
+        row_letter__range=range_serializer.get_row_range(),
+    )
+    if racks.count() == 0:
+        failure_message = "No racks were deleted because no racks within" + \
+            " the specified range exist."
+        return JsonResponse(
+            {"failure_message": failure_message},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    elif racks.count() < range_serializer.get_num_racks_in_range():
+        failure_message = "No racks were deleted because some racks within" + \
+            " the specified range do not exist."
+        return JsonResponse(
+            {"failure_message": failure_message},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    unempty_racks = []
+    for rack in racks:
+        if ITInstance.objects.filter(rack=rack.id).count() > 0:
+            unempty_racks.append(rack.row_letter + str(rack.rack_num))
+    if len(unempty_racks) > 0:
+        failure_message = "No racks were deleted because the following " + \
+            "racks contain instances: " + ", ".join(unempty_racks)
+        return JsonResponse(
+            {"failure_message": failure_message},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    else:
+        try:
+            racks.delete()
```

Comment:
You can call this on a list of models? Wild

Reply:
p what? p wild